packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -380,8 +382,26 @@ class AppSizeController {
 
     Map<String, dynamic> diffMap;
     if (oldFile.isAnalyzeSizeFile && newFile.isAnalyzeSizeFile) {
+      var oldFileJson = oldFile.data as Map<String, dynamic>;
+      var newFileJson = newFile.data as Map<String, dynamic>;
+
+      if (_hasDeferredInfo(oldFileJson) || _hasDeferredInfo(newFileJson)) {
+        _isDeferredApp.value = deferredLoadingSupportEnabled;
+
+        if (!_hasDeferredInfo(oldFileJson)) {
+          oldFileJson = <String, dynamic>{
```

Comment:
Let's pull this logic out into a private method (eg, `_wrapInArtificialRoot`) which takes the json file as a parameter. That way we don't have to repeat the code here and below. 

I think we also have to rename the first node from `Root` to `Main` before we wrap it, ie:

```
          oldFileJson['n'] = 'Main';
          oldFileJson = <String, dynamic>{
            'n': deferredRootName,
            'children': [oldFileJson]
          };
```